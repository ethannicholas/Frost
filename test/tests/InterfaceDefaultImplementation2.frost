interface Test1 {
    method foo()
}

interface Test2 : Test1 {
    @override
    @default
    method foo() {
        Console.printLine("Test2.foo")
    }
}

class Test3 : Test2 {
}

class Test4 : Test1 {
    @override
    method foo() {
        Console.printLine("Test4.foo")
    }
}

class Test5 : Test2 {
    @override
    method foo() {
        Console.printLine("Test5.foo")
    }
}

method main() {
    Test3().foo()
    Test4().foo()
    Test5().foo()

    var t:Test1
    t := Test3()
    t.foo()
    t := Test4()
    t.foo()
    t := Test5()
    t.foo()
}