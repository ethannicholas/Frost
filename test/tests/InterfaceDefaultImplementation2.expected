Test2.foo
Test4.foo
Test5.foo
Test2.foo
Test4.foo
Test5.foo
